torch/_inductor/codegen/cuda/cutlass_lib_extensions/gemm_operation_extensions.py

```diff
@@ -0,0 +1,181 @@
+# To be imported only after cutlass_utils.try_import_cutlass() has been called.
+import enum
+
+from cutlass_library import *  # type: ignore[import]  # noqa: F401, F403
+from cutlass_gemm_operation import *  # type: ignore[import]  # noqa: F401, F403
+
+
+# copied / modified from original at
+# https://github.com/NVIDIA/cutlass/blob/8783c41851cd3582490e04e69e0cd756a8c1db7f/tools/library/scripts/gemm_operation.py#L658
+# to support EVT similar to
+# https://github.com/NVIDIA/cutlass/blob/8783c41851cd3582490e04e69e0cd756a8c1db7f/examples/49_hopper_gemm_with_collective_builder/49_collective_builder.cu#L315C69-L315C69  # noqa: B950
+class EmitGemmUniversal3xInstanceWithEVT:
+    """Responsible for emitting a CUTLASS 3.x template definition"""
+
+    def __init__(self, operation_suffix=""):
+        self.operation_suffix = operation_suffix
+        self.includes = [
+            "cutlass/cutlass.h",
+            "cutlass/gemm/gemm.h",
+            "cutlass/numeric_types.h",
+            "cutlass/gemm/kernel/gemm_universal.hpp",
+            "cutlass/gemm/collective/collective_builder.hpp",
+            "cutlass/epilogue/collective/collective_builder.hpp",
+        ]
+        self.builtin_epilogue_functor_template = """
+    ${epilogue_functor}<
+      ${element_c},
+      ${epilogue_vector_length},
+      ${element_accumulator},
+      ${element_epilogue}
+    >
+"""
+        self.gemm_template = """
+using EpilogueScheduleType = ${epilogue_schedule};
+static_assert(cute::is_same_v<EpilogueScheduleType, cutlass::epilogue::TmaWarpSpecialized> ||
+         cute::is_same_v<EpilogueScheduleType, cutlass::epilogue::TmaWarpSpecializedCooperative>,
+        "Epilogue visitor trees are currently only supported by the TMA warp-specialized epilogue");
+static constexpr auto RoundStyle = cutlass::FloatRoundStyle::round_to_nearest;
+using ElementAcc = ${element_accumulator};
+${epilogue_functor};
+using ${operation_name}_epilogue =
+  typename cutlass::epilogue::collective::CollectiveBuilder<
+    ${arch}, ${opcode_class},
+    cute::Shape<cute::_${tile_shape_m}, cute::_${tile_shape_n}, cute::_${tile_shape_k}>,
+    cute::Shape<cute::_${cluster_m},cute::_${cluster_n},cute::_${cluster_k}>,
+    cutlass::epilogue::collective::EpilogueTileAuto,
+    ${element_accumulator}, ${element_epilogue},
+    ${element_c}, ${layout_c}, ${align_c},
+    ${element_d}, ${layout_d}, ${align_d},
+    EpilogueScheduleType,
+    ${operation_name}_epilogue_functor
```

Comment:
And here we pass the epilogue_functor as template argument, which did not happen in the original code.